pkgs/dart_mcp_server/test/test_harness.dart

```diff
@@ -384,8 +357,32 @@ Future<String> _getDTDUri(TestProcess dtdProcess) async {
   return dtdUri;
 }
 
-typedef ServerConnectionPair =
-    ({ServerConnection serverConnection, DartMCPServer? server});
+/// Reads DTD uri from the [dtdProcess] output.
+Future<String> _getDTDSecret(TestProcess dtdProcess) async {
```

Comment:
If you start dtd with the --machine flag you can parse the output as json instead of reading stdout lines. Example: https://github.com/flutter/devtools/blob/master/packages/devtools_shared/test/helpers/helpers.dart/#L18. Then you could possibly merge this method with the one above to return a DTD info object as a pattern